### PR TITLE
executor: fix incorrect result when hash agg spill is triggered

### LINKED ISSUE #55290: hash agg parallel spill get wrong result
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
for a simple query like 
```
select count(*), id from t group by id
```
if parallel spill is triggered, it will return wrong result.

### PR DESCRIPTION
**Pull request by xzhangxian1008**
### What problem does this PR solve?

Issue Number: close #55290

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix the issue that getting incorrect result when hash agg spill is triggered
```

### CODE REVIEW

File: pkg/executor/aggregate/agg_spill_test.go [hunk 1]
```diff
@@ -191,22 +199,44 @@ func buildHashAggExecutor(t *testing.T, ctx sessionctx.Context, child exec.Execu
 	schema := expression.NewSchema(childCols...)
 	groupItems := []expression.Expression{childCols[0]}
 
-	aggFirstRow, err := aggregation.NewAggFuncDesc(ctx.GetExprCtx(), ast.AggFuncFirstRow, []expression.Expression{childCols[0]}, false)
+	var err error
+	var aggFirstRow *aggregation.AggFuncDesc
```
[the remark this answers is not in the document]
Comment (xzhangxian1008) [reply pkg/executor/aggregate/agg_spill_test.go:203]: > miss aggAvg

avg function is inconvenient to be tested in ut, i will test it in endless.

File: pkg/executor/aggregate/agg_spill_test.go [hunk 2]
```diff
@@ -372,15 +419,17 @@ func TestGetCorrectResult(t *testing.T) {
 		wg.Done()
 	}()
 
+	hardLimitBytesNum := int64(6000000)
+	initCtx(ctx, newRootExceedAction, hardLimitBytesNum, 256)
+
 	aggExec := buildHashAggExecutor(t, ctx, dataSource)
-	for i := 0; i < 5; i++ {
+	for i := 0; i < 1; i++ {
```
[the remark this answers is not in the document]
Comment (xzhangxian1008) [reply pkg/executor/aggregate/agg_spill_test.go:426]: > why change this?

Mistake, i will fix it.

**Review by XuHuaiyu — APPROVED** [review]

**Review by windtalker — APPROVED** [review]
lgtm

**Pull request merged**